vochain/transaction/account_tx.go

```diff
@@ -239,3 +239,42 @@ func (t *TransactionHandler) SetAccountInfoTxCheck(vtx *vochaintx.Tx) error {
 	}
 	return nil
 }
+
+// DelSikTxCheck checks if a delete sik tx is valid
+func (t *TransactionHandler) DelSikTxCheck(vtx *vochaintx.Tx) error {
+	if vtx == nil || vtx.Signature == nil || vtx.SignedBody == nil || vtx.Tx == nil {
+		return ErrNilTx
+	}
+	tx := vtx.Tx.GetSetSik()
+	if tx == nil {
+		return fmt.Errorf("invalid transaction")
+	}
+
+	bAddress := tx.GetAddress()
```

Comment:
This is wrong, you cannot trust the Address available in the Protobuf model is correct.

Actually the address should not be a field, instead you should extract the address from the Signature.

Please check initializeSignedVote() function. You need to do something similar.

Reply:
Let's ensure no other TxCheck method works like this... either we have a huge security hole.